R/ResultFlagsIndependent.R

```diff
@@ -1021,48 +1022,117 @@ TADA_FindPotentialDuplicates <- function(.data, dist_buffer = 100){
       }else{
-        .data$TADA.DuplicateID = NA
-        .data$TADA.ProbableDuplicate = "N"
+        .data$TADA.BetweenOrgDupGroupID = "Not a duplicate"
+        .data$TADA.ProbableBetweenOrgDuplicate = "N"
         print("No duplicate results detected. Returning input dataframe with duplicate flagging columns set to N.")
       }
     }else{ # if no site duplicates detected
-      .data$TADA.DuplicateID = NA
-      .data$TADA.ProbableDuplicate = "N"
+      .data$TADA.BetweenOrgDupGroupID = "Not a duplicate"
+      .data$TADA.ProbableBetweenOrgDuplicate = "N"
       print("No duplicate results detected. Returning input dataframe with duplicate flagging columns set to N.")
     }
   }else{ # if no result/org duplicates detected
-    .data$TADA.DuplicateID = NA
-    .data$TADA.ProbableDuplicate = "N"
+    .data$TADA.BetweenOrgDupGroupID = "Not a duplicate"
+    .data$TADA.ProbableBetweenOrgDuplicate = "N"
     print("No duplicate results detected. Returning input dataframe with duplicate flagging columns set to N.")
   }
   
   .data = TADA_OrderCols(.data)
   
   return(.data)
-}
\ No newline at end of file
+}
+
+#' Identify Potential Duplicate Data Uploads Within An Organization (UNDER ACTIVE DEVELOPMENT)
+#' 
+#' Identifies data records uploaded by the same organization with the same date,
+#' time, monitoring location, activity type, characteristic name, fraction,
+#' taxonomic name, depth columns, and result value and flags as potential
+#' duplicates. However, it is at the discretion of the data user to determine if
+#' the data records are unique or represent overlap that could cause issues in
+#' the data analysis.
+#' 
+#' @param .data TADA dataframe
+#' @param handling_method Defaults to 'none'. When input is 'none', function
+#'   flags groups of potentially duplicated results, but makes no decisions on
+#'   which (if any) should be selected/excluded. Alternatively, user may
+#'   populate handling_method = 'pick_one' and function will randomly select one
+#'   result in each group and flag all others as duplicates.
+#' 
+#' @return The same input TADA dataframe with additional columns: a
+#'   TADA.WithinOrgDupGroupID column indicating whether a result is part of a
+#'   group that shares the same date, time, location, characteristic, etc. and
+#'   TADA.ResultSelectedWithinOrg, which defaults to TRUE (all results
+#'   selected), unless the user specifies the handling method as 'pick_one', in
+#'   which case there exists one result in each duplicate group where
+#'   TADA.ResultSelectedWithinOrg = TRUE and for all other results in the group
+#'   TADA.ResultSelectedWithinOrg = FALSE.
+#' 
+#' @export
+#' 
+
+TADA_FindPotentialDuplicatesWithinOrg <- function(.data, handling_method = 'none'){
```

Comment:
If combined: 

handling_method could be randomly select 1, choose 1 monitoringlocationID of the two (single org dup), or choose 1 organizationID of the two (multiple org dup).

The distance buffer would be helpful for both of these. Within a single org, there may be a retired and current site that may not be at exactly the same point, but have duplicates of the historical data.

Reply:
We can discuss further, but my initial reaction is that the first suggestion would be difficult to execute. I'm not sure how someone would pick a single monitoring location or organization when every single duplicate group has potentially unique comparisons (i.e. duplicate groups comparing between orgs will not necessarily always have the same two or more organizations within the duplicate group; every duplicate group has a set of unique monitoring locations....not sure how you'd confidently pick one naming convention that applies to all sites in each duplicate group.)

Re: second suggestion, I think there is a low likelihood that a data uploader would upload the same data result twice for a current and a retired monitoring location. I'm sure it has happened, but I wasn't sure it was worth creating a function for it. Do you think the need is great enough to craft this additional function? It might be easier to just use the TADA_IDNearbySites function and custom group them in the second module.